torch/distributed/_spmd/api.py

```diff
@@ -51,3 +73,119 @@ def forward(self, *args: Tuple[object], **kwargs: Dict[str, object]) -> object:
 
         assert self._compiled_m is not None
         return self._compiled_m(*args, **kwargs)
+
+
+class Override(ABC):
+    @abstractmethod
+    def replacement(self, orig_submodule: torch.nn.Module) -> torch.nn.Module:
+        pass
+
+    @abstractmethod
+    def transform(
+        self, gm: fx.GraphModule, context: Dict[fx.Node, DTensor]
+    ) -> fx.Graph:
+        pass
+
+
+OVERRIDES: Dict[Type[Any], Override] = {}
+
+
+def _register_override(typ: Any, override: Override) -> None:
```

Comment:
I wonder if it's better to pass overrides as arguments to _compile as opposed to registering them. I think most overrides are instances that needs to be customized via configs.